packages/account/src/providers/transaction-request/transaction-request.ts

```diff
@@ -105,6 +107,8 @@ export abstract class BaseTransactionRequest implements BaseTransactionRequestLi
   outputs: TransactionRequestOutput[] = [];
   /** List of witnesses */
   witnesses: TransactionRequestWitness[] = [];
+  /** Base asset ID */
+  baseAssetId: string = ZeroBytes32;
```

Comment:
Left the default in as it minimises breaking changes and provides a better DX for users on local test net. All docs and docs snippets will reference fetching and setting this value.

If we want to fetch a sensible default value, we'll need to pass in a provider to the transaction request.

Reply:
If it's not set and the user does not own the `baseAssetId` then the tx will fail validation as the user does not have enough of the required coins to fund the transaction. Or they don't set it, and the base asset ID is `ZeroBytes32`, and the user does have the asset then tx will succeed (equal to current behaviour).

Reply:
I share the frustration, and I also understand this would rely on TX validation (runtime), but it might be better to fail earlier (compile time). Of course, it's a DX trade-off, but something tells me we should not favor local test net for this decision.

Reply:
If we are going to have `baseAssetId` on `TransactionRequest` we need to make it an optional property, and this needs to be set later on.

Setting this early as `ZeroBytes32` can be misleading

Reply:
Went through the design and DX implications with @nedsalk earlier and attempting to mitigate breaking changes. There isn't much value in moving the breaking change from the constructor to the function signatures. And making it optional would require checks where it is used.

The real value would be in making a transaction request agnostic from the idea of a base asset ID. Asset IDs should just be passed, if it should be a base asset ID, then that is up to the user. This is a design decision right now to improve DX. However the only function that we can see that requires knowing the base asset ID specifically is `addMessageInput` which uses the base asset to create a change output.

Given the above is out of scope here, and this approach is low risk as requires the user to pass the base asset at tx instantiation, I am going to resolve this conversation.

Reply:
@danielbate I don't follow. A `MessageCoin` has the property `assetId` set on it. 

Why TransactionRequest needs to know the BaseAsset when adding an `OutputChange` for it?

Reply:
Responded [here](https://github.com/FuelLabs/fuels-ts/pull/1976#issuecomment-2065849482).

Reply:
Tx request no longer requires a base asset ID. https://github.com/FuelLabs/fuels-ts/pull/1976/commits/1239a4bbaac8a320ced1c66eeab51224269e0d8d

Reply:
@danielbate can you update the PR description?